httpx/config.py

```diff
@@ -166,6 +168,11 @@ def _create_default_ssl_context(
         if ssl.HAS_NPN:  # pragma: no cover
             context.set_npn_protocols(http_versions.alpn_identifiers)
 
+        if hasattr(context, "keylog_filename"):
+            keylogfile = os.environ.get("SSLKEYLOGFILE")
+            if keylogfile and not sys.flags.ignore_environment:
```

Comment:
I believe there’s a trust_env option on the client for this kind of check — is it available here/would it make sense to use it here as well?

Reply:
I think it makes sense for us to ignore this variable when we use `trust_env=False`. I guess that'll have to be piped into `SSLConfig`.

Reply:
Gave it a shot in the latest commit. I'm a bit unsure about the default behavior of `trust_env` in `SSLConfig`: currently it defaults to `trust_env=None`, but I wonder whether it should imitate the behavior of `BaseClient` when it's set to `None`.

Reply:
On re-reading yes, we probably want to have the same default in both places.

What would we say about setting the default to True though, instead of None and then defaulting that to True? It would be simpler, and I don’t think we need to support the idea of « I passed a None value to get the default behavior » here. Thoughts @sethmlarson?